File: code_samples/java/OopEncapsulation.java
/**
 * Oop: Encapsulation
 * AI/ML Training Sample
 */
public class OopEncapsulation {
    
    private String data;
    
    public OopEncapsulation() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        OopEncapsulation instance = new OopEncapsulation();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}

File: code_samples/swift/oop_encapsulation.swift
import Foundation

/// Oop: Encapsulation
/// AI/ML Training Sample

class Encapsulation {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = Encapsulation()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")
